src/distPlus/symbolic/SymbolicDist.re

```diff
@@ -47,12 +47,31 @@ type dist = [
   | `Cauchy(cauchy)
   | `Triangular(triangular)
   | `ContinuousShape(continuousShape)
-  | `Float(float)
+  | `Float(float) // Dirac delta at x. Practically useful only in the context of multimodals.
 ];
 
-type pointwiseAdd = array((dist, float));
+type integral = float;
+type cutoffX = float;
+type operation = [
+    | `AddOperation
+    | `SubtractOperation
+    | `MultiplyOperation
+    | `DivideOperation
+    | `ExponentiateOperation
+];
 
-type bigDist = [ | `Simple(dist) | `PointwiseCombination(pointwiseAdd)];
+type distTree = [
+    | `Simple(dist)
+    | `Combination(distTree, distTree, operation)
+    | `PointwiseSum(distTree, distTree)
+    | `PointwiseProduct(distTree, distTree)
```

Comment:
I think I'd like to make this pretty clean in this refactor, given we are at it.

I'm not sure why PointwiseSum, PointwiseProduct, and VerticalScaling, are handled different from the `Combination operations. It seems more elegant to me to just include these in the list of `operations`. There are a few ways this could be done; it could be split into three sublists; one for standard functions, one for pointwise, one for "vertical," and each could be defined. 

I'll spend more time thinking here.

Reply:
I do like the idea of a nested operations type. Dumping all the cases into one giant switch statement is an option, but I think the code may still be easier to understand if we break it down into two or three separate evaluation functions ... let me think about that some more. Do you have a clear preference?

Reply:
One good similar example is the way errors are handled in functional languages. Often there are a few types of error, each with its own subtypes. 

I like the idea of the sublists. Maybe something along the lines of:

type standardOperation = [
  `Sum(distTree, distTree)
  `Product(distTree,distTree),
... 
]

type pointwiseOperation = [
  `Sum(distTree, distTree)
  `Product(distTree,distTree),
]

type pointwiseOperation = [
  `Sum(distTree, distTree)
  `Product(distTree,distTree),
]

## This applies the mean of the scaleBy parameter to the distTree parameter for some kind of mathematical operation. 

scaleParams = {
  distTree: distTree,
  scaleBy: distTree
} 

type scaleOperation = [
  `Product(scaleParams),
  `Add(scaleParams),
]

``type operation = [
  `StandardOperation(standardOperation),
  `PointwiseOperation(pointwiseOperation),
  `ScaleOperation(scaleOperation),
]

Reply:
Nice, that's a good breakdown. Can you elaborate on how the `ScaleOperation` would work, especially the `Add` subtype? What's being added there?

Reply:
It would be weird. The idea is that it would take the full distribution, and shift it up (along the y axis) by n. Multiplying takes each y-value and multiplies it, adding would take each y-value and adds to it. 

Maybe ``yShift`` may be a better name.

Reply:
If I add 1 everywhere, the result would add up to infinity, and can't be normalized anymore. Same with exp/log. I suppose e.g. squaring every value would work though

Reply:
Exp/log would apply to the y values, not the x values. 

Also, I'd agree that plussing/minusing would make it non-normalizable. I think that hypothetically would be ok; like, we could throw an error if one were to try to normalize these. That said, this could be a bit much for now, and I don't see use cases for people wanting this anytime soon.

Reply:
Right but exp(0) = 1. Perhaps `y = exp(a*y) - 1` would work? Anyways I see where you were going with `scaleOperation`, thanks!

Reply:
Ah, good point, you're right. Maybe it's possible that parts of the distribution could be "undefined", but probably not at this point.